src/penumbra.cpp

```diff
@@ -162,135 +155,142 @@ float Penumbra::fetchPSSA(unsigned surfaceIndex) {
+      throw PenumbraException(
+          fmt::format("Transparent surface index, {}, does not exist. Cannot calculate PSSA.",
+                      transparentSurfaceIndices[0]),
+          *(penumbra->logger));
     }
   } else {
-    showMessage(MSG_ERR, "Cannot calculate interior PSSAs without defining at least one "
-                         "transparent surface index."); // TODO format string
+    throw PenumbraException(
+        "Cannot calculate interior PSSAs without defining at least one transparent surface index.",
+        *(penumbra->logger));
   }
   return pssas;
 }
 
-int Penumbra::renderScene(unsigned surfaceIndex) {
- if (penumbra->checkSurface(surfaceIndex)) {
-   penumbra->context.showRendering(surfaceIndex, penumbra->sun.getView());
-   return PN_SUCCESS;
- } else {
-   showMessage(MSG_ERR,
-               "Surface index, X, does not exist. Cannot render scene."); // TODO format string
-   return PN_FAILURE;
- }
+void Penumbra::renderScene(unsigned surfaceIndex) {
+  if (penumbra->checkSurface(surfaceIndex)) {
+    penumbra->context.showRendering(surfaceIndex, penumbra->sun.getView());
+  } else {
+    throw PenumbraException(
+        fmt::format("Surface index, {}, does not exist. Cannot render scene.", surfaceIndex),
+        *(penumbra->logger));
+  }
 }
 
-int Penumbra::renderInteriorScene(std::vector<unsigned> transparentSurfaceIndices,
-                                  std::vector<unsigned> interiorSurfaceIndices) {
- if (transparentSurfaceIndices.size() > 0) {
-   if (penumbra->checkSurface(transparentSurfaceIndices[0])) {
-     for (auto &transSurf : transparentSurfaceIndices) {
-       if (!penumbra->checkSurface(transSurf)) {
-         showMessage(
-             MSG_ERR,
-             "Transparent surface index, X, does not exist. Cannot calculate PSSA."); // TODO format string
-       }
-     }
-     for (auto &intSurf : interiorSurfaceIndices) {
-       if (penumbra->checkSurface(intSurf)) {
-         penumbra->context.showInteriorRendering(transparentSurfaceIndices, intSurf, penumbra->sun.getView());
-       } else {
-         showMessage(
-             MSG_ERR,
-             "Interior surface index, X, does not exist. Cannot calculate PSSA."); // TODO format
-                                                                                   // string
-         return PN_FAILURE;
-       }
-     }
-     return PN_SUCCESS;
-   } else {
-     showMessage(
-         MSG_ERR,
-         "Transparent surface index, X, does not exist. Cannot calculate PSSA."); // TODO format
-                                                                                  // string
-     return PN_FAILURE;
-   }
- } else {
-   showMessage(MSG_ERR, "Cannot calculate interior PSSAs without defining at least one "
-                        "transparent surface index."); // TODO format string
-   return PN_FAILURE;
- }
+void Penumbra::renderInteriorScene(std::vector<unsigned> transparentSurfaceIndices,
```

Comment:
Could these be passed as const ref (as in the `calculatePSSA` functions) and then the for loop uses auto const too, as above?
Then it would make sense to declare some of the helper functions const too (`checkSurface`, etc.), but maybe another PR.